maven.grammar/src/org/netbeans/modules/maven/hyperlinks/HyperlinkProviderImpl.java

```diff
@@ -453,5 +215,331 @@ private FileObject getPath(FileObject parent, String path) {
         }
         return parent.getFileObject(path);
     }
+    
+    private class PomParserRunnable implements Runnable {
+
+        private final PomHyperlinkInfo hyperLinkInfo;
+        private final Document document;
+        private final int offset;
+
+        public PomParserRunnable(PomHyperlinkInfo hyperLinkInfo, Document document, int offset) {
+            this.hyperLinkInfo = hyperLinkInfo;
+            this.document = document;
+            this.offset = offset;
+        }
+        
+        @Override
+        public void run() {
+            TokenHierarchy th = TokenHierarchy.get(document);
+            TokenSequence<XMLTokenId> xml = th.tokenSequence(XMLTokenId.language());
+            xml.move(offset);
+            xml.moveNext();
+            Token<XMLTokenId> token = xml.token();
```

Comment:
it's often better to check moveNext()'s value than the result of token(), but both work.